### DISCUSSION TITLE
Something is breaking tab navigation

### DISCUSSION THREAD

**Discussion by ddbrierton** [LazyVim/LazyVim#735] [Q&A] [answered]
I think this needs to be a bug report, but I’m not sure whether it’s a bug with LazyVim or one of the plugins, and if the latter which one.

These are my tests.

1. Start nvim with no configuration: `nvim -u /dev/null`
2. Create some tabs with `:tabnew`
3. Navigate between them using `:tabprevious` and `:tabnext`.
4. Confirm it all works as expected. ✅

Now, test with a pristine LazyVim installation.

1. Move all custom lua configuration files to a folder outside of `~/.config/nvim` leaving just the pristine versions of LazyVim’s `init.lua`, `lua/config/lazy.lua` and `lua/plugins/example.lua`.
2. Start `nvim`.
3. Again, create some tabs with `:tabnew`  and navigate between them using `:tabprevious` and `:tabnext`, and also with the keyboard shortcuts, `<space><tab> [` and `<space><tab> ]`.
4. Confirm it all works as expected. ✅.
5. Restart `nvim` without restoring your previous session.
6. This time open a bunch of tabs using `<space> f f` or by opening them from `neo-tree` (`<space> e` select file with `j` or `k` and open with `<enter`).
7. Navigating between tabs either with the keyboard shortcuts of with `:tabprevious` and `:tabnext` no longer works (or sometimes sort of works, but some tabs are unreachable. ❌

Can anyone else confirm? It seems very strange that no one else has noticed this.

I’m afraid that this is well beyond my meagre nvim skills to track down any further.

**Comment by dpetka2001** [comment, marked as answer]
`<space>ff` and neo-tree opening file with `<enter>` doesn't open files in tabs. It opens them in buffers. So I think it's normal you can't use `tabprevious`/`tabnext`. Except if you open files from Telescope (`<space>ff`) with `ctrl-t` which you didn't mention doing, so I guess that you're opening them with `<enter>` normally.

**Reply by ddbrierton** [reply]
Oh — I guess this might be the correct answer. (And I will mark it as such if it is.) In which case I guess I don’t understand what a tab is in vim. I mean, I know what a buffer is and what a window is in vim. Files are always opened in buffers. Always. A window is a view on a buffer. And a tab is, I thought, a collection of windows.

I mean, what happens on screen when you do `:tabnew` is identical to what happens when you open a file from neo-tree or Telescope by hitting enter. It doesn’t replace the contents of your current window, it appear to create a new tab.

So I’m just confused now ...

**Reply by loiccoyle** [reply]
FYI: the "tabs" at the top of the nvim window are actually buffers.

**Reply by dpetka2001** [reply]
`tabnew` creates indeed a new tab, but you can see the other buffers as well. If you close a tab with a buffer in it, the buffer will not close. It will remain being seen from another tab. You would have to delete the buffer (`:bdelete`) for it to go away and not see it any more. You could say tabs are kind of just different layouts. You can also configure each tab with its own current working directory, which could make your workflow like grouping similar files depending on your working directory. Or maybe some other way I can't think of right now.

**Reply by ddbrierton** [reply]
Thank you @loiccoyle and @dpetka2001 for the clarification. Indeed I now see that I can navigate between the “tabs” with previous and next buffer, `H` and `L`. So the things which look exactly like what everyone else call “tabs” are not in fact tabs, except when they are because they were created with `:newtab` 😊.

Thank you both for you help.

**Reply by dpetka2001** [reply]
When you split vertically or horizontally you create new windows. So, one tab might have windows split horizontally and another tab might have windows split vertically. In that essence, tabs are indeed just collections of windows. That's why I mentioned you could see them as kind of different layouts.

**Reply by ddbrierton** [reply]
Yes, thanks @dpetka2001 . Vim’s concept of buffer and window are familiar to me, they’re the same in Emacs too. Vim’s idea of a tab is new to me. I’ll get used to it, but I was thrown by the fact that nvim+LazyVim uses a visual representation for buffers that would be described as tabs elsewhere.

I’m still a little confused though, because I thought Vim could have an open buffer that wasn’t being displayed in any window whatsoever.

**Reply by dpetka2001** [reply]
The visual representation is the product of `bufferline`, the plugin Lazyvim uses for showing bufferlist in tabline. You could try something more plain like [nvim-tabline](https://github.com/crispgm/nvim-tabline) which only shows tabs on the tabline and in each tab it only shows the currently active buffer. That means you would have to do `<space>,` to show which buffers you have opened and navigate that way through them (since it doesn't display all the buffers on the tabline like `bufferline`). It might seem more cumbersome as a workflow, but it all depends on your needs and what you're accustomed to.
